oxo.py

```diff
@@ -1,26 +1,72 @@
 class OxoBoard:
     def __init__(self):
         """ The initialiser. Initialise any fields you need here. """
-        raise NotImplementedError("TODO: implement __init__")
+        #sets the number of rows and collems there are
+        self.rows = 3
+        self.columns = 3
+
+        self.numberOfTiles = self.rows * self.columns
+
+        self.board = []
+        for row in range(self.rows):
+            self.board += [[0] * self.columns]
+        print self.board
+        #raise NotImplementedError("TODO: implement __init__")
 
     def get_square(self, x, y):
         """ Return 0, 1 or 2 depending on the contents of the specified square. """
-        raise NotImplementedError("TODO: implement get_square")
+        #retuns the state of the square the mouse is over
+        return self.board[y][x]
+        #raise NotImplementedError("TODO: implement get_square")
 
     def set_square(self, x, y, mark):
         """ If the specified square is currently empty (0), fill it with mark and return True.
             If the square is not empty, leave it as-is and return False. """
-        raise NotImplementedError("TODO: implement set_square")
+        #if the number which is retuned above is 0 then it but the player mark on the square
+        if self.get_square(x, y) == 0:
+            self.board[y][x] = mark
+            return True
+        else:
+            return False
+        #raise NotImplementedError("TODO: implement set_square")
 
     def is_board_full(self):
         """ If there are still empty squares on the board, return False.
             If there are no empty squares, return True. """
-        raise NotImplementedError("TODO: implement is_board_full")
+        # see if the board is filled
+        tiles_filled = 0
+
+        for row in self.board:
+            for item in row:
+                if item != 0:
+                    tiles_filled += 1
+                    if tiles_filled == self.numberOfTiles:
+                        return True
```

Comment:
Not the most efficient method (e.g. you could `return False` as soon as you see a 0), but it works